lib/workos/directory_sync.rb

```diff
@@ -175,15 +175,15 @@ def get_group(id)
           ),
         )
 
-        JSON.parse(response.body)
+        ::WorkOS::DirectoryGroup.new(response.body)
```

Comment:
🤔 Hmm, that's a good point - I was focusing on accessing the individual attributes via `[]`, but that's a very good point, there's lots of other methods on a Hash. I suppose we could create a Hash under the covers and delegate everything to it, though it seems a bit messy... but probably worth it? We still wouldn't have consistency if somebody had written a `group.is_a?(Hash)` check for some reason... but we can probably live with that. 🤔 I wonder if I could make `DirectoryUser` and `DirectoryGroup` inherit from `Hash`?

Reply:
Inheriting from Hash _seems_ to work... at least the tests are happy, and I did add tests to verify Hash-style access. The only gotcha so far is that we have to overwrite any individual Hash methods if we want them to give the deprecation warning... so far I've only done that for `[]`, so `dig`, `fetch`, etc won't give the warning 🤷‍♂️.

On the bright(-er) side, `group.is_a?(Hash)` does return true now, even if `group.class` now returns `DirectoryGroup`.

@mthadley, does this seem reasonable to you? Are there any other cases you can think of that would be worth checking/adding to our tests? 🙇‍♂️

Reply:
So, I was thinking about this a little more, and what if we just didn't deprecate the Hash-style access? With the classes inhering from `Hash`, I think the implementation becomes reasonably clean, and I don't see quite as compelling of reason to deprecate it... What do you think?

If we do decide to go ahead and deprecate, then 👍 to your suggestion ☝️ - that seems to me like it would work well!

Reply:
👋 @maxdeviant, are you suggesting we don't inherit from `Hash` at all, even temporarily, or don't do it long term - i.e. continue with the deprecation plan as originally proposed?

As far as exposing the API surface... I feel like that is, unfortunately, the problem - currently, we return a Hash, so users are getting access to that entire API surface. I'd initially gone for the quick fix of just adding Hash-style attribute access, but it seems likely that [at least some] customers are using other Hash methods, so it might cause issues if we break that :( ... so, at that point, we either inherit from Hash, or just delegate all the methods to it - I'm not crazy about either option, but at least inheritance is cleaner 🤷‍♂️. 💭's?

Reply:
> 👋 @maxdeviant, are you suggesting we don't inherit from Hash at all, even temporarily, or don't do it long term - i.e. continue with the deprecation plan as originally proposed?

Apologies, I should have been clearer.

I think it's a fine temporary solution (since we are currently exposing a `Hash` and as such we don't know what operations consumers may be doing on them).

However, I think that for the long-term design of the SDK we shouldn't be inheriting from `Hash`.

Reply:
Sounds good, thanks @maxdeviant, and I'm on board with that - I'll go ahead and add the `DeprecatedHashWrapper` that @mthadley had suggested (🤔 and maybe add a `to_h` method, so users can request a Hash), and we can then deprecate it.